### PR TITLE
Use snprintf

### LINKED ISSUE #5650: 'sprintf' is deprecated in coords_profile.cc:52:2: on MacOS 13
Widelands fails to compile with 

```
FAILED: src/map_io/CMakeFiles/map_io.dir/coords_profile.cc.o 
...XcodeDefault.xctoolchain/usr/bin/c++ -DASIO_STANDALONE -I ...  -DDEBUG -g -fsanitize=address -fno-omit-frame-pointer -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion -Wno-exit-time-destructors -Wno-global-constructors -Wno-padded -Wno-sign-conversion -Wno-missing-noreturn -Wno-switch-enum -Wno-weak-vtables -Wno-unreachable-code -Wno-documentation -Qunused-arguments -Wint-to-void-pointer-cast -Winconsistent-missing-override -Werror -Wno-error=poison-system-directories -Wno-error=disabled-macro-expansion -MD -MT src/map_io/CMakeFiles/map_io.dir/coords_profile.cc.o -MF src/map_io/CMakeFiles/map_io.dir/coords_profile.cc.o.d -o src/map_io/CMakeFiles/map_io.dir/coords_profile.cc.o -c /Users/klaus/develop/widelands/master/src/map_io/coords_profile.cc
/Users/klaus/develop/widelands/master/src/map_io/coords_profile.cc:52:2: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
        sprintf(buffer, "%i %i", value.x, value.y);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 error generated.
[61/322] Building CXX object src/map_io/CMakeFiles/map_io.dir/map_buildingdata_packet.cc.o
```

**Version:**
 - OS: macOS 13.0 , Apple Silicon
 - Widelands Version: 1.2~git25917 (1549bed@master)
 
**Additional context**

I assume there is a reason why sprintf() is deprected ...

### PR DESCRIPTION
**Pull request by klaus-halfmann**
Fixes #5650.

was the only sprintf (But usage was uncritical)

### CODE REVIEW

File: src/map_io/coords_profile.cc [hunk 1]
```diff
@@ -48,8 +48,10 @@ Coords parse_coords(const std::string& name, const char* const coords, const Ext
 }  // namespace
 
 void set_coords(const std::string& name, const Coords& value, Section* section) {
-	char buffer[sizeof("-32769 -32769")];
-	sprintf(buffer, "%i %i", value.x, value.y);
+	const int size = sizeof("-32769 -32769");
+	char buffer[size];
+	snprintf(buffer, size, "%i %i", value.x,
+	         value.y);  // No need to check result as result should awlwayy fit
```
Comment (Noordfrees) [inline src/map_io/coords_profile.cc]:
```suggestion
	         value.y);  // No need to check result as result should always fit
```

File: src/map_io/coords_profile.cc [hunk 2]
```diff
@@ -48,8 +48,10 @@ Coords parse_coords(const std::string& name, const char* const coords, const Ext
 }  // namespace
 
 void set_coords(const std::string& name, const Coords& value, Section* section) {
-	char buffer[sizeof("-32769 -32769")];
-	sprintf(buffer, "%i %i", value.x, value.y);
+	const int size = sizeof("-32769 -32769");
```
Comment (Noordfrees) [inline src/map_io/coords_profile.cc]: Can this be made `constexpr`?

Comment (klaus-halfmann) [reply src/map_io/coords_profile.cc]: Should not make a difference in the end, but will do this.

**Review by Noordfrees — APPROVED** [review]

**Comment by klaus-halfmann** [comment]
@Noordfrees wee need to merge this into 1.1, too so I can 
compile https://github.com/widelands/widelands/pull/5656
I will fix this locally for now, should I push this to the 1.1. branch directly?

**Comment by klaus-halfmann** [comment]
Pushed this directly now

**Pull request merged**